[bryantjandra] iP

broyantt opened the pull request:
# Robert

> “*If you spend too much time thinking about a thing, you’ll never get it done.*” – Bruce Lee

**Robert** is your personal companion for managing tasks. It’s:

- **Fast** (written in Java)
- *Simple* (easy to learn)
- ~~Expensive~~ **FREE** forever!

Here’s how you can start using it:

1. **Download** the JAR from [this release page](https://github.com/bryantjandra/ip/releases).
2. **Double-click** the JAR file, or run `java -jar robert.jar`.
3. **Add** your tasks (`todo`, `deadline`, `event`).
4. Let it manage your tasks while you focus on bigger things! :smiley:

### Features

- [x] Manage tasks (todo, events, deadlines)
- [x] Show tasks in a list
- [ ] *Reminders* (coming soon)

If you are a Java programmer, you can use it to practice Java, especially with inline code like `System.out.println("Hello, I'm Robert")`. Additionally, here’s the **main method** used for storing the tasks locally!

```java
public static void main(String[] args) {
    new Robert("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

broyantt committed:
Rename chatbot to Robert and add greeting and farewell

broyantt committed:
Add echo functionality and handle bye command

broyantt committed:
Add storage and list functionality to store and display user input

broyantt committed:
Add Task class and integrate mark/unmark commands

broyantt committed:
Add support for todos, deadlines, and events with inheritance to handle different types of tasks

broyantt committed:
Add input/output redirection technique to semi-automate testing

broyantt committed:
Add exception handling for user input validation

broyantt committed:
Refactor codebase to add javadoc comments

broyantt committed:
Add delete command and switch to ArrayList to store tasks

broyantt committed:
Use an enum-based command approach

broyantt committed:
Implement saving of tasks upon shutdown and loading upon startup

broyantt committed:
Implement saving of tasks upon shutdown and loading upon startup

broyantt committed:
Merge branch-Level-7 to master

broyantt committed:
Implement date parsing for deadlines

broyantt committed:
Merge branch-Level-8 to master

broyantt committed:
Refactor code into Ui, Storage, TaskList, and Parser classes

broyantt committed:
Merge branch-A-MoreOOP into master

broyantt committed:
Refactor code into multiple packages

broyantt committed:
Merge branch-A-Packages into master

broyantt committed:
Merge add-gradle-support into master

broyantt committed:
Integrate Gradle and fix checkstyle errors

broyantt committed:
Add JUnit tests for several nontrivial public methods

broyantt committed:
Merge branch-A-JUnit to master

broyantt committed:
Create fat JAR file as a release

broyantt committed:
Fix any typos and errors in JavaDoc comments

broyantt committed:
Fix code to comply with coding standard

broyantt committed:
Add find command to search tasks with keywords

broyantt committed:
Merge branch-A-JavaDoc into master

broyantt committed:
Merge branch-A-CodingStandard into master

broyantt committed:
Merge branch-Level-9 into master

kimseunghyun-kr commented on src/main/java/robert/task/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDate byDate;
```
```
dueDate
```

1. I like how you used byDate, but i think the above seems to be a better name

kimseunghyun-kr commented on src/main/java/robert/Robert.java:
```diff
@@ -0,0 +1,299 @@
+    public static void main(String[] args) {
+        new Robert("data/tasks.txt").run();
+    }
+}
```
1. The class seems a bit too long. Why not separate it into different classes i.e)
```
InputHandler.java
```
2. good naming convention followed, classes are nouns and methods are verbs.

kimseunghyun-kr commented on src/test/java/robert/parser/ParserTest.java:
```diff
@@ -0,0 +1,50 @@
+     * Tests if parse("") returns CommandType.UNKNOWN.
+     */
+    @Test
+    public void parse_emptyString_returnUnknown() {
+        CommandType result = Parser.parse("");
+        assertEquals(CommandType.UNKNOWN, result);
+    }
+
+    /**
+     * Tests if parse("todo read book") → CommandType.TODO.
+     */
+    @Test
+    public void parse_todoCommand_returnTodo() {
+        CommandType result = Parser.parse("todo read book");
+        assertEquals(CommandType.TODO, result);
+    }
+
+    /**
+     * Example: parse("bye") → BYE.
+     */
+    @Test
+    public void parse_byeCommand_returnBye() {
+        CommandType result = Parser.parse("bye");
+        assertEquals(CommandType.BYE, result);
+    }
+
+    /**
+     * Tests that parse does not return null even for random strings.
+     */
+    @Test
+    public void parse_randomInput_returnNotNull() {
+        CommandType result = Parser.parse("randomstuff");
+        assertNotNull(result);
+    }
+}
```
i like that you followed test code conventions and clearly named your test code as (action)_(returns). however, i would like to suggest that you include exception testing in your work

kimseunghyun-kr commented on src/main/java/robert/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
i like that you followed naming conventions and used is{var_name} for your boolean variable. very nice

kimseunghyun-kr reviewed (commented):
LGTM, expand on test cases in the future, to include exception testing. looking for future updates

broyantt committed:
Fix formatting error of gradle file for checkstyle

broyantt committed:
Merge branch-A-CheckStyle to master

karthu0301 commented on src/main/java/robert/Robert.java:
```diff
@@ -0,0 +1,299 @@
+import robert.ui.Ui;
+
+/**
+ * Main class of the Robert chatbot application.
```
Descriptions of all the methods look good, but maybe consider splitting up this class into a few more classes to separate the functionality and for better readability of code.

karthu0301 commented on src/main/java/robert/parser/Parser.java:
```diff
@@ -0,0 +1,22 @@
+package robert.parser;
```
Consider moving some functionality from Robert.java to over here perhaps.

karthu0301 commented on src/main/java/robert/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+}
```
Overall, your Task classes LGTM, good job on keeping them as simple as possible!

broyantt committed:
Add GUI for robert chatbot

broyantt committed:
Merge branch-Level-10 to master

broyantt committed:
Add assertions to enforce developer assumptions

Previously, the code relied on runtime checks or simply threw exceptions
when users provided invalid data. However, some invariants (e.g., a
non-null file path, or a valid index in TaskList) are meant to be
programmer-level guarantees rather than user-facing checks.

Failing these assumptions indicates a bug in the code rather than normal
user error. Without explicit assertions, such violations can go unnoticed
until they cause unexpected behavior elsewhere.

Let's add assert statements in key methods such as Storage, TaskList,
and command-parsing utilities. This approach clarifies critical
expectations, documents them in the code, and helps catch developer
mistakes early in testing.

broyantt committed:
Merge pull request #1 from bryantjandra/branch-A-Assertions

Enforce developer assumptions using assertions

broyantt committed:
Refactor Storage class to reduce duplication and improve readability

Storage.java has duplicated logic in load() and save() methods and uses
arrowhead-style nesting when parsing lines. This makes the code harder to
read and maintain.

Duplicated logic in load() complicates scanning file lines, while
inline conditionals in save() obscure the overall flow. Readers also
have to scan through repeated "if-else" blocks in both methods.

Let's extract the line-parsing logic into a parseTaskLine() helper
method. Similarly, let's move the task-to-string conversion into a
convertTaskToString() method. Doing so keeps load() and save() concise
and easier to follow.

Both methods now focus on high-level steps, making it clear how tasks
are read from and written to disk without getting bogged down by
low-level details. This approach also helps reduce code duplication,
minimize nesting, and improve overall maintainability.

broyantt committed:
Merge pull request #2 from bryantjandra/branch-A-CodeQuality

Refactor Storage class to reduce duplication and improve readability

broyantt committed:
Add C-Sort extension to sort deadlines by date

broyantt committed:
Merge branch-C-Sort to master

broyantt committed:
Improve GUI for better readability and UX

broyantt committed:
Merge branch-BetterGUI to master

broyantt committed:
Enhance chatbot personality with a polite butler theme

broyantt committed:
Merge branch-Personality to master

broyantt committed:
Improve error handling for deadlines and tasks

broyantt committed:
Merge branch-MoreErrorHandling to master

broyantt committed:
Add JUnit testing for most classes

broyantt committed:
Merge branch-MoreTesting to master

broyantt committed:
Add AI markdown file describing the help of AI

broyantt committed:
Updated README.md

broyantt committed:
Replace UI with unique user images

broyantt committed:
Add screenshots for user guide

broyantt committed:
Update sample data

broyantt committed:
Make Robert instance final to enforce immutability

broyantt committed:
Update JavaDoc comment

broyantt committed:
Fix gramatical error in JavaDoc comment
